File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples-js/ColumnsListByDatabaseMax.js
const { SqlManagementClient } = require("@azure/arm-sql");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to List database columns
 *
 * @summary List database columns
 * x-ms-original-file: specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/examples/ColumnsListByDatabaseMax.json
 */
async function filterDatabaseColumns() {
  const subscriptionId =
    process.env["SQL_SUBSCRIPTION_ID"] || "00000000-1111-2222-3333-444444444444";
  const resourceGroupName = process.env["SQL_RESOURCE_GROUP"] || "myRG";
  const serverName = "serverName";
  const databaseName = "myDatabase";
  const schema = ["dbo"];
  const table = ["customer", "address"];
  const column = ["username"];
  const orderBy = ["schema asc", "table", "column desc"];
  const options = {
    schema,
    table,
    column,
    orderBy,
  };
  const credential = new DefaultAzureCredential();
  const client = new SqlManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.databaseColumns.listByDatabase(
    resourceGroupName,
    serverName,
    databaseName,
    options,
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

File: specification/sql/resource-manager/Microsoft.Sql/SQL/stable/2025-01-01/examples-java/ColumnsListByDatabaseMax.java

import java.util.Arrays;

/**
 * Samples for DatabaseColumns ListByDatabase.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-01-01/ColumnsListByDatabaseMax.json
     */
    /**
     * Sample code: Filter database columns.
     * 
     * @param manager Entry point to SqlServerManager.
     */
    public static void filterDatabaseColumns(com.azure.resourcemanager.sql.SqlServerManager manager) {
        manager.serviceClient().getDatabaseColumns().listByDatabase("myRG", "serverName", "myDatabase",
            Arrays.asList("dbo"), Arrays.asList("customer", "address"), Arrays.asList("username"),
            Arrays.asList("schema asc", "table", "column desc"), null, com.azure.core.util.Context.NONE);
    }
}
